### PR TITLE
Add preloading of comments

### LINKED ISSUE #5537: My Sites:Preload site content when user navigates to the my sites page
When the user navigates to the my sites page and taps on any item for the first time the item should open to pre loaded content. To respect cellular data usage we should probably only do this on wifi.

### PR DESCRIPTION
**Pull request by nheagy**
Fixes #5537 - preloads comments when on wi-fi.

## To test:
_Note: there is a cache timer that is persisted to disk. Preloading only fires once per blog per five minutes. This will affect testing._

### On LTE/3G:
There shouldn't be any preloading.

1. View the details page for a blog with new comments. The network activity indicator should _not_ appear.
2. Tap the 'Comments' row
3. The network activity indicator should appear, and new comments loaded.

### On Wi-fi:
There should be preloading according to the following scenarios:
#### First view of a blog in 5 or more minutes
1. Once network activity indicator is gone on blog details view, new comments should appear on comments view.
2. The comments view should not automatically hit the network itself.

#### Subsequent view of a blog in under 5 minutes
1. No comments preloading on blog details view. *Since pages, posts, and stats are preloaded, the network activity indicator cannot be used to judge*
2. Comments view itself will not automatically load new comments. Pull-to-refresh will, however.

#### When viewing the comments view before preloading has completed
1. Open a blog detail view for a blog outside its 5 minute cache limit. Immediately open the comments view, before the preloading has completed.
2. The comments view should not fire a second network request.
3. Once the preloading has completed, the comments view should refresh with new comments.

Let me know if these instructions are confusing or not helpful.

Needs review: @koke
Welcome back!

### CODE REVIEW

**Comment by koke** [comment]
My main concern if we start preloading content is that we should make sure to prioritize and not duplicate requests:

- Do not duplicate: if I go to blog details and immediately go to comments (or any other preloaded section), it's going to fire two requests to sync comments.
- Prioritize: preloading requests should not block the network for other requests. (See [previous conversation](https://github.com/wordpress-mobile/WordPress-iOS/pull/4001#issuecomment-120965315)). 
- I'd also say don't preload if the latest sync was recent enough. If I go to view comments (which syncs when it load) and go back, there's no need to "preload" comments again.

**Comment by nheagy** [comment]
I've added a cache timeout so that comments are only automatically loaded once every 5 minutes. Pull-to-refresh still works to force reload.

@koke, @aerych — thoughts? If this is good, I'll add similar timeout to the other preloading once merged.

File: WordPress/Classes/ViewRelated/Comments/CommentsViewController.m [hunk 1]
```diff
@@ -446,6 +450,31 @@ - (void)syncContentEnded
...
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:CommentServiceFinishedSyncingNotification object:nil];
+}
+
+- (void)syncNotificationReceived:(NSNotification *)notification
+{
+    NSDictionary *notificationInfo = notification.userInfo;
+    NSManagedObjectID *blogObjectID = notificationInfo[@"objectID"];
+    if ([blogObjectID isEqual:self.blog.objectID])
+    {
+        [self.tableViewHandler refreshTableView];
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Comments/CommentsViewController.m]: I don't think you need the notification, as the fetched results controller will automatically refresh the table as soon as the new data is saved.

Comment (nheagy) [inline WordPress/Classes/ViewRelated/Comments/CommentsViewController.m]: Yup, it's still working fine without it. Not sure why I thought I needed it… 🤔 but the code is a touch simpler now 👍

**Pull request merged**